app/controllers/posts_controller.rb

```diff
@@ -4,54 +4,40 @@ class PostsController < ApplicationController
   respond_to :html, :json
   respond_to :rss, only: %i(index show)
 
-  # GET /posts
-  # GET /posts.json
-  # GET /posts.rss
+  responders :flash
+
   def index
     @author = Member.find_by(slug: params[:author])
-    @posts = posts
+    @posts = @posts.where author: @author if @author.present?
+    @posts = @posts.order(created_at: :desc).includes(:author, comments: :author).paginate(page: params[:page])
     respond_with(@posts)
   end
 
-  # GET /posts/1
-  # GET /posts/1.json
-  # GET /posts/1.rss
   def show
-    @post = Post.includes(:author, comments: :author).find(params[:id])
```

Comment:
This is likely more optimised than the load_and_authorise_resource version, as it would generate a left join instead of multiple DB calls